File: week-05/yc_3986_good_word.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());
        int good = 0;

        for(int i = 0; i < n; i++) {
            char[] word = br.readLine().trim().toCharArray();
            char[] stack = new char[word.length];
            int top = -1;

            for(char c : word) {
                if (top >= 0 && stack[top] == c) {
                    top--;
                }
                else {
                    stack[++top] = c;
                }
            }
            if(top == -1) good++;
        }
        System.out.println(good);
    }
}

File: week-05/yc_3986_good_word.cs
using System;
using System.IO;

class Program
{
	static void Main(string[] args)
	{
		StreamReader sr = new StreamReader(Console.OpenStandardInput());
		string line = sr.ReadLine();
		int n = int.Parse(line!.Trim());
		int good = 0;

		for(int i = 0; i < n; i++)
		{
			var word = sr.ReadLine()!.Trim().AsSpan();
			char[] stack = new char[word.Length];
			int top = -1;

			for(int j = 0; j < word.Length; j++)
			{
				char c = word[j];
				if(top >= 0 && stack[top] == c)
				{
					top--;
				}
				else
				{
					stack[++top] = c;
				}
			}
			if(top == -1) good++;
		}
		Console.WriteLine(good);
	}
}
